flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/FileSystemJobResultStoreTest.java

```diff
@@ -0,0 +1,285 @@
+import org.apache.flink.util.TestLoggerExtension;
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+/** Tests for the {@link FileSystemJobResultStore}. */
+@ExtendWith(TestLoggerExtension.class)
+public class FileSystemJobResultStoreTest {
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private static final JobResultEntry DUMMY_JOB_RESULT_ENTRY = createDummyJobResultEntry();
+
+    private FileSystemJobResultStore fileSystemJobResultStore;
+
+    @TempDir File temporaryFolder;
+
+    @BeforeEach
+    public void setupTest() throws IOException {
+        Path path = new Path(temporaryFolder.toURI());
+        fileSystemJobResultStore = new FileSystemJobResultStore(path.getFileSystem(), path, false);
+    }
+
+    /** Tests that the base directory is created on Filesystem Job Result Store initialization. */
+    @Test
+    public void testBaseDirectoryCreationOnResultStoreInitialization() throws Exception {
+        final File tempDir = new File(temporaryFolder.getPath() + "/new-temp-dir");
+        final Path tempPath = new Path(tempDir.getPath());
+        assertThat(tempDir).doesNotExist();
+
+        fileSystemJobResultStore =
+                new FileSystemJobResultStore(tempPath.getFileSystem(), tempPath, false);
+        assertThat(tempDir).exists().isDirectory();
+    }
+
+    /** Tests that adding a JobResult to the JobResultStore results in a Dirty JobResultEntry. */
+    @Test
+    public void testStoreDirtyJobResult() throws Exception {
+        fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
+        assertThat(fileSystemJobResultStore.hasJobResultEntry(DUMMY_JOB_RESULT_ENTRY.getJobId()))
+                .isTrue();
+
+        Set<JobResult> dirtyResults = fileSystemJobResultStore.getDirtyResults();
+        assertThat(dirtyResults.stream().map(JobResult::getJobId).collect(Collectors.toList()))
+                .containsExactlyInAnyOrder(DUMMY_JOB_RESULT_ENTRY.getJobId());
+        assertThat(getCleanResultIdsFromFileSystem()).isEmpty();
+
+        assertThat(expectedDirtyFile(DUMMY_JOB_RESULT_ENTRY)).exists().isFile().isNotEmpty();
+    }
+
+    @Test
+    public void testStoreDirtyJobResultTwice() throws IOException {
+        fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
+        assertThat(expectedDirtyFile(DUMMY_JOB_RESULT_ENTRY)).exists().isFile().isNotEmpty();
```

Comment:
```suggestion
```
nit: That is already tested in `testStoreDirtyJobResult` and is out-of-scope for this test case. We're interested in the second `createDirtyResult` call here. Same goes for the file creation after `markAsClean` is called.